grafana-plugin/src/containers/UserSettings/parts/tabs/GoogleCalendar/GoogleCalendar.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react';
+
+import { Button, Divider, HorizontalGroup, InlineSwitch, VerticalGroup } from '@grafana/ui';
+import { observer } from 'mobx-react';
+
+import { Block } from 'components/GBlock/Block';
+import { Text } from 'components/Text/Text';
+import { WithConfirm } from 'components/WithConfirm/WithConfirm';
+import { GSelect } from 'containers/GSelect/GSelect';
+import { WithPermissionControlTooltip } from 'containers/WithPermissionControl/WithPermissionControlTooltip';
+import GoogleLogo from 'icons/GoogleLogo';
+import { Schedule } from 'models/schedule/schedule.types';
+import { UserHelper } from 'models/user/user.helpers';
+import { ApiSchemas } from 'network/oncall-api/api.types';
+import { useStore } from 'state/useStore';
+import { UserActions } from 'utils/authorization/authorization';
+
+const GoogleCalendar: React.FC<{ id: ApiSchemas['User']['pk'] }> = observer(({ id }) => {
+  const { userStore, scheduleStore } = useStore();
+
+  const user = userStore.items[id];
+  const [googleCalendarSettings, setGoogleCalendarSettings] = useState(user?.google_calendar_settings);
+  const [showSchedulesDropdown, setShowSchedulesDropdown] = useState(
+    user.google_calendar_settings?.specific_oncall_schedules_to_sync.length > 0
+  );
+
+  const handleShowSchedulesDropdownChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.checked;
+    setShowSchedulesDropdown(value);
+
+    if (!value) {
+      handleSchedulesChange([]);
+    }
+  };
+
+  useEffect(() => {
+    if (user) {
+      setGoogleCalendarSettings(user.google_calendar_settings);
+    }
+  }, [user]);
+
+  const handleSchedulesChange = (value) => {
+    setGoogleCalendarSettings((v) => ({ ...v, specific_oncall_schedules_to_sync: value }));
+
+    userStore.updateCurrentUser({
+      google_calendar_settings: { ...googleCalendarSettings, specific_oncall_schedules_to_sync: value },
+    });
+  };
+
+  return (
+    <VerticalGroup>
+      <Block bordered style={{ width: '100%' }}>
+        <VerticalGroup>
+          <VerticalGroup spacing="none">
+            <Text.Title level={5}>Google Calendar</Text.Title>
+            <Text type="secondary">
+              Connect personal calendar to sync PTOs and OOO events with your OnCall schedules.
```

Comment:
```suggestion
              Connect your Google calendar to read Out of Office events, and autogenerate Shift Swap Request(s) when we notice you are out of office, but also on-call for one or more OnCall schedules.
```

Reply:
I suggest to use this wording:
Connect personal Google calendar
Connection allows OnCall to read your Out of Office events and autogenerate Shift Swap Request

Reply:
👍 (would suggest removing the word "personal", to me it suggests non-work related Calendar)

Reply:
I've updated the wording in https://github.com/grafana/oncall/pull/4067/commits/fe4d35eac0b3753f092ed0139a3cbc7b9121ae6c